hatchery/iam.go

```diff
@@ -59,19 +79,80 @@ func (creds *CREDS) taskRole(userName string) (*string, error) {
 			  `),
 		}
 
+		createTaskRole, err := svc.CreateRole(createTaskRoleInput)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create TaskRole: %s", err)
+		}
+
 		_, err = svc.AttachRolePolicy(&iam.AttachRolePolicyInput{
-			PolicyArn: policy.Policy.Arn,
+			PolicyArn: &policyArn,
 			RoleName:  aws.String(userToResourceName(userName, "pod")),
 		})
 		if err != nil {
 			return nil, fmt.Errorf("failed to attach RolePolicy: %s", err)
 		}
-		createTaskRole, err := svc.CreateRole(createTaskRoleInput)
-		if err != nil {
-			return nil, fmt.Errorf("failed to create TaskRole: %s", err)
-		}
 
 		return createTaskRole.Role.Arn, nil
 	}
 
 }
+
+const ecsTaskExecutionRoleName = "ecsTaskExecutionRole"
+const ecsTaskExecutionPolicyArn = "arn:aws:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy"
+const ecsTaskExecutionRoleAssumeRolePolicyDocument = `{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "",
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "ecs-tasks.amazonaws.com"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}`
```

Comment:
do we need to put some comment to explain what this is doing and why it is needed?

Reply:
Good call! Added a comment